src/com/itacademy/eshop/product/Cash.java

```diff
@@ -0,0 +1,41 @@
+package com.itacademy.eshop.product;
+
+import com.itacademy.eshop.shop.ShoppingCart;
+
+public abstract class Cash extends Payment {
+
+    private String currency;
+    private double change;
+
+    public Cash(String paymentMethod, double amount, String currency, double change) {
+        super(paymentMethod, amount);
+        this.currency = currency;
+        this.change = change;
+    }
+
+    public String getCurrency() {
+        return currency;
+    }
+
+    public void setCurrency(String currency) {
+        this.currency = currency;
+    }
+
+    public double getChange() {
+        return change;
+    }
+
+    public void setChange(double change) {
+        this.change = change;
+    }
+
+    @Override
+    public void processPayment(ShoppingCart shoppingCart) {
```

Comment:
since you are only using shoppingCart.getTotalPrice, it would make more sense to only pass that value instead of the entire shoppingCart object to this method